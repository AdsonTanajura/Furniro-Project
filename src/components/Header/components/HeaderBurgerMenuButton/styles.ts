import styled from 'styled-components';

export const BurgerMenuButton = styled.button`
  display: flex;
  flex-direction: column;
  gap: 4px;
  background: none;
  border: none;
  cursor: pointer;

  span {
    width: 24px;
    height: 3px;
    background-color: black;
    border-radius: 2px;
  }

  @media (min-width: 780px) {
    display: none;
  }
`;
