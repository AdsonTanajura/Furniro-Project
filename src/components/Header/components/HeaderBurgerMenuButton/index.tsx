import HeaderBurgerMenuButtonProps from './Interfaces/HeaderBurgerMenuButtonProps';
import { BurgerMenuButton } from './styles';

const HeaderBurgerMenuButton = ({
  toggleMenu,
}: HeaderBurgerMenuButtonProps) => {
  return (
    <BurgerMenuButton onClick={toggleMenu}>
      <span />
      <span />
      <span />
    </BurgerMenuButton>
  );
};

export default HeaderBurgerMenuButton;
