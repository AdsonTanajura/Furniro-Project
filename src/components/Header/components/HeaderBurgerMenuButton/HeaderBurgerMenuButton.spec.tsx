import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HeaderBurgerMenuButton from '.';

describe('<HeaderBurgerMenuButton />', () => {
  const mockSetIsMenuOpen = jest.fn();
  const mockIsMenuOpen = false;
  it('should render all elements correctly', () => {
    const { getAllByRole, container } = render(
      <HeaderBurgerMenuButton
        setIsMenuOpen={mockSetIsMenuOpen}
        isMenuOpen={mockIsMenuOpen}
      />
    );
    const buttons = getAllByRole('button');

    expect(buttons.length).toBeGreaterThan(0);
    expect(buttons.length).toBeLessThanOrEqual(1);
    buttons.forEach((button) => expect(button).toBeInTheDocument());
    buttons.forEach((button) => expect(button).toBeVisible());

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should return true when the button is clicked', async () => {
    const { getByRole, container } = render(
      <HeaderBurgerMenuButton
        setIsMenuOpen={mockSetIsMenuOpen}
        isMenuOpen={mockIsMenuOpen}
      />
    );
    const button = getByRole('button');

    expect(mockSetIsMenuOpen).not.toHaveBeenCalled();

    await userEvent.click(button);
    await waitFor(() => expect(mockSetIsMenuOpen).toHaveBeenCalled());
    await waitFor(() => expect(mockSetIsMenuOpen).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(mockSetIsMenuOpen).toHaveBeenCalledWith(true));

    expect(container.firstChild).toMatchSnapshot();
  });
});
