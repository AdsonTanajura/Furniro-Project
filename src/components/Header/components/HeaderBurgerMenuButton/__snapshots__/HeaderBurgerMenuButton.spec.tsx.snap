// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBurgerMenuButton /> should render all elements correctly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  background: none;
  border: none;
  cursor: pointer;
}

.c0 span {
  width: 24px;
  height: 3px;
  background-color: black;
  border-radius: 2px;
}

@media (min-width: 780px) {
  .c0 {
    display: none;
  }
}

<button
  class="c0"
>
  <span />
  <span />
  <span />
</button>
`;

exports[`<HeaderBurgerMenuButton /> should return true when the button is clicked 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 4px;
  background: none;
  border: none;
  cursor: pointer;
}

.c0 span {
  width: 24px;
  height: 3px;
  background-color: black;
  border-radius: 2px;
}

@media (min-width: 780px) {
  .c0 {
    display: none;
  }
}

<button
  class="c0"
>
  <span />
  <span />
  <span />
</button>
`;
